metricproviders/plugin/rpc/rpc.go

```diff
@@ -154,17 +152,13 @@ func (g *MetricsPluginRPC) GetMetadata(metric v1alpha1.Metric) map[string]string
 // the requirements of net/rpc
 type MetricsRPCServer struct {
 	// This is the real implementation
-	Impl MetricsPlugin
+	Impl MetricProviderPlugin
 }
 
-// NewMetricsPlugin is the receiving end of the RPC call running in the plugin executable process (the server), and it calls the
+// InitPlugin is the receiving end of the RPC call running in the plugin executable process (the server), and it calls the
 // implementation of the plugin.
-func (s *MetricsRPCServer) NewMetricsPlugin(args interface{}, resp *types.RpcError) error {
-	initArgs, ok := args.(*InitMetricsPluginAndGetMetadataArgs)
-	if !ok {
-		return fmt.Errorf("invalid args %s", args)
-	}
-	*resp = s.Impl.NewMetricsPlugin(initArgs.Metric)
+func (s *MetricsRPCServer) InitPlugin(args interface{}, resp *types.RpcError) error {
+	*resp = s.Impl.InitPlugin()
```

Comment:
Is this the function that will crash the container if it is unable to initialize the plugin? If so, isn't it better to return an error and avoid Argo Rollouts to fail during bootstrap?

Reply:
No this is actually net/rpc code that is ran within the plugin itself, the plugin host this function the controller calls it via rpc [code](https://github.com/argoproj/argo-rollouts/pull/2573/files#diff-f219b498990aac5ae38a8c5d388a3517bf44e0698ef28aa840186851ba9df508R60) the `resp` value here becomes the value that get's passed to the controller from the plugin.